src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,337 @@
+
+    def get(self, setting: Any) -> Any:
+        """
+        Helper function for getting experiment parameters with a basic check
+        to see if a specific setting is valid.
+        """
+        if self.experiment.Exists(setting):
+            value = self.experiment.GetValue(setting)
+        else:
+            value = []
+            logger.error(f'Invalid setting: {setting}.')
+        return value
+
+    def load_experiment(self, value: str) -> None:
+        self.experiment.Load(String(value))
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationAttachDate, False)
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationAttachTime, False)
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationAttachIncrement, True)
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationIncrementNumber, Int32(0))
+
+    def file_setup(self) -> None:
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationBaseFileName, str(uuid.uuid4()))
+    
+    #NOTE: Thw while loop in here will not be needed once you fix the "FileNameGeneration" problem.
+    def start_acquisition_and_wait(self) -> None:
+        """
+        Starts the acquisition process and waits until it is completed before carrying on.
+        """
+        acquisition_time_seconds = self.get(lf.AddIns.CameraSettings.ShutterTimingExposureTime) / 1000.0
+        num_frames = self.get(lf.AddIns.ExperimentSettings.AcquisitionFramesToStore)
+        self.experiment.Acquire()
+
+        sleep(num_frames * acquisition_time_seconds) 
+        
+        while self.experiment.IsRunning:
+            sleep(0.1)  # checking if the experiment is still running
+
+    def process_acquired_data(self) -> np.ndarray:
+        """
+        Processes the most recently acquired data and returns it as a numpy array.
+        """
+        last_file = self._application.FileManager.GetRecentlyAcquiredFileNames()[0]
+        image_dataset = self._application.FileManager.OpenFile(last_file, FileAccess.Read)
+
+        if image_dataset.Regions.Length == 1:
+            if image_dataset.Frames == 1:
+                return self._process_single_frame(image_dataset)
+            else:
+                return self._process_multiple_frames(image_dataset)
+        else:
+            raise QT3Error(f"LightField FileManager OpenFile error. Unsupported value for Regions.Length: {image_dataset.Regions.Length}.Should be == 1")
+
+    def _process_single_frame(self, image_dataset: Any) -> np.ndarray:
+        """
+            **Single Frame**:
+            - Data is returned as a 2D array representing raw counts from each pixel.
+            - A vertical section (spanning 400 pixels) represents a range for wavelength averaging.
+        """
+        frame = image_dataset.GetFrame(0, 0)
+        return np.reshape(np.fromiter(frame.GetData(), dtype='uint16'), [frame.Width, frame.Height], order='F')
+
+    def _process_multiple_frames(self, image_dataset: Any) -> np.ndarray:
```

Comment:
Why Any as a type hint?

Reply:
Maybe there is something I am missing in the documentation but if you look at the comment above I don't really understand what type that is... I assume a list/vector of some kind

Reply:
Actually, I don't think it is a List....